Mathlib/Data/List/Basic.lean

```diff
@@ -412,6 +412,14 @@ theorem append_left_injective (t : List α) : Injective fun s ↦ s ++ t :=
 
 #align list.map_eq_append_split List.map_eq_append_split
 
+lemma length_append_append :
+    (l₁ ++ l₂ ++ l₃).length = l₁.length + l₂.length + l₃.length := by
+  rw [length_append, length_append]
+
+lemma reverse_append_append :
+    (l₁ ++ l₂ ++ l₃).reverse = l₃.reverse ++ l₂.reverse ++ l₁.reverse := by
+  rw [reverse_append, reverse_append, append_assoc]
```

Comment:
These don't strike me as good lemmas; the bar needs to be higher than "needs more than one existing lemma to prove", as that set is unbounded!

Reply:
I think the analogous lemma I'm thinking of would be `op_mul_mul : op (a * b * c) = op (c * b * a)`, which mathlib also doesn't have (which I claim is evidence that we don't want to go down the road of adding this kind of lemma).

Reply:
What is `op` in your example?

Reply:
`MulOpposite.op`

Reply:
I see your point.

Reply:
My opinion is that in both cases, it should depend on the number of uses in mathlib in near future.

On one hand, these lemmata don't provide any value in the sense that anybody who would be able to use them also would be able to easily prove them themselves.

On the other hand, existence of these lemmata in my project makes my proofs shorter; and you are the first one who complains that my proofs are too long / not enough modular. If you forbid `List.length_append_append` et al., it will get even worse.

Reply:
I'm happy to be overruled here by another reviewer

Reply:
They pop up very often when working with languages, not only in the `List.append_singleton_append_inj_of_nmem` proof. But of course, I can do everything without them if preferable.

Reply:
It is preferable.

Reply:
Also there are proofs with a state where `List.reverse_append` matches the goal in several places but `List.reverse_append_append` matches only one place in the goal, leading to a greater simplification when rewriting with the latter.

Reply:
You can rewrite with `(reverse_append _ _).trans (congr_arg (· ++ _) $ reverse_append _ _)` if you want to.

Reply:
OK removed.